cpp/INIReader.cpp

```diff
@@ -56,6 +56,16 @@ long INIReader::GetInteger(const string& section, const string& name, long defau
     return end > value ? n : default_value;
 }
 
+INI_API long long INIReader::GetBigInteger(const std::string& section, const std::string& name, long long default_value) const
```

Comment:
Two things:

1) I think if we keep these type `long long` we should call these `GetLongLongInteger` and `GetLongLongUnsigned`, as they're clearer than "big" (even if a bit, um, long). In my mind "big integer" usually means arbitrarily large like Go's `math/big.Int` types.
2) Is it common in modern C++ to use "long long", or is it more common to use the specific width types like `int64_t` and `uint64_t`? I've been out of the C++ loop for a while.

Reply:
I've had a quick look around, and it seems `(u)int64_t` is fairly common in C++ codebases. Plus, I think fixed sizes are more sane than "at least this long" sizes. And given that we both prefer it, let's go with that for these new functions, and call them `GetInteger64` and `GetUnsigned64` as you suggest.

I presume you'll have to `#include <cstdint>` to get these?

Reply:
Alright, I updated the code to use those method names and fixed-width types.

These types were already accessible with <map> included, but it's probably best to include <cstdint> anyways, so I have done that.

Reply:
You do know that `(u)int64_t` is less portable than `(unsigned) long long`? The latter is guaranteed to exist in all conforming C++11-and-later implementations while the former is entirely optional. While this might not be the most relevant on desktop, it could be an issue for embedded uses where toolchains don't implement every optional C++ feature. Using `int64_t` is not simply a matter of coding style: it's for when your code is so frail that it relies on exact size and overflow/underflow behaviour.

Reply:
@Clownacy Maybe you're right. It looks like they've caused on issue on an old version of GCC: https://github.com/benhoyt/inih/issues/152

That said, I was with you up until your last sentence: it seems to me using fixed-width integers is likely to make code *less* frail, because the width is specified concretely and exactly, rather than "may be this big, may overflow at this point, but may not, you'll never know".